### PR TITLE
breaking the code and adding new functionality

### PR DESCRIPTION
**Pull request by Draganek**
I split the code a bit and started creating new functionality in drink list

### CODE REVIEW

File: frontend/drink_list_picker.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        name.className = "drink_name"
+        div.className = "drink_element";
+        img.className = "drink_img";
+        img.src = element.link;
+        div.append(img);
+        div.append(name);
+        drink_holder.append(div);
+    }
+}
+login_user();
```
Comment (kfron) [inline frontend/drink_list_picker.js]: File should end with an empty line, add an empty line after this line.

File: frontend/drink_list_picker.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+
+
+async function login_user() {
+    let drink_holder = document.getElementById("drink_list_show");
+
```
Comment (kfron) [inline frontend/drink_list_picker.js]: No consecutive empty lines, remove excess empty lines

File: frontend/drink_list_picker.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+
```
Comment (kfron) [inline frontend/drink_list_picker.js]: Remove excess empty lines

File: frontend/drink_list_picker.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+async function login_user() {
```
Comment (kfron) [inline frontend/drink_list_picker.js]: This doesn't look like the right name for this function

File: frontend/index.html [hunk 5]
```diff
@@ -6,14 +6,17 @@
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 <title>Dragan Bar</title>
 <link rel="stylesheet" href="style.css">
+<script src="login_menu.js" defer></script>
 <script src="script.js" defer></script>
+<script src="drink_list_picker.js" defer></script>
+
```
Comment (kfron) [inline frontend/index.html:2]: remove empty line

File: frontend/index.html [hunk 6]
```diff
@@ -129,46 +132,9 @@ <h1 style="padding-left: 15px; color: white;">
...
-        Siemano
-      </a>
-    </div>
-
-
-
-
+  <div class="drink_label" id="drink_list_show">
   </div>
+
```
Comment (kfron) [inline frontend/index.html]: What is this empty line for?

File: frontend/login_menu.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+// ======================================== Logowanie do strony ======================================== //
```
Comment (kfron) [inline frontend/login_menu.js]: Comments this long are not more helpful than normal comments, and can even be more annoying on smaller screens or when the code part of IDE is smaller in width than your editor.
In other words, don't use the equal signs to make the comment "nicer"

File: frontend/login_menu.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+let login_button = document.getElementById("submit_button");
+let user_login2 = document.getElementById("user_login");
```
Comment (kfron) [inline frontend/login_menu.js]: Why user_login2 and not just user_login?

File: frontend/login_menu.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+let user_password2 = document.getElementById("user_password");
```
Comment (kfron) [inline frontend/login_menu.js]: Same for this one, why the '2' at the end?

File: frontend/login_menu.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+let login_status = document.getElementById("login_info");
+
+login_button.addEventListener("click", () => login_user(user_login,user_password));
+
+async function login_user(login, password){
+    let response = await fetch(`users.json`);
+    let obiekt = await response.json();
```
Comment (kfron) [inline frontend/login_menu.js]: Stick to english

File: frontend/login_menu.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        if (login.value === "admin" && password.value === "admin"){
+            login_status.value = "Chłop co wpisał admin admin xdddd"
+            return;
+        }
+        if (login.value === "" || password.value === ""){
+            login_status.value = "Uzupełnij pola logowania"
+            return;
+        }
+        for (let check of obiekt.account){
+
```
Comment (kfron) [inline frontend/login_menu.js]: Remove empty line

File: frontend/login_menu.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+        }
+        if (login.value === "" || password.value === ""){
+            login_status.value = "Uzupełnij pola logowania"
+            return;
+        }
+        for (let check of obiekt.account){
+        
+            if (check.login === login.value){
+                if(check.password === password.value){
+                    login_status.value = "Logowanie..."
```
Comment (kfron) [inline frontend/login_menu.js]: There is a lot of hard-coded values used for `login_status.value`. What do you think about creating an enum for them?
There are no native enums in JavaScript, but you could do something like that:
```
const LoginStatuses = {
  IN_PROGRESS: 'Logowanie',
  MISSING_INPUT: 'Uzupełnij pola logowania',
  INVALID_PASSWORD: 'Błędne hasło',
  INVALID_USER: 'Nie ma takiego użytkownika',
  EASTER_EGG:'Chłop co wpisał admin admin xdddd'
}
```

Then you can assign the value as follows: `login_status.value = LoginStatuses.IN_PROGRESS` etc.

File: frontend/login_menu.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+            return;
+        }
+        if (login.value === "" || password.value === ""){
+            login_status.value = "Uzupełnij pola logowania"
+            return;
+        }
+        for (let check of obiekt.account){
+        
+            if (check.login === login.value){
+                if(check.password === password.value){
```
Comment (kfron) [inline frontend/login_menu.js]: This if inside another if could be written in a better way.
I would do something like this:
```
if(check.login !== login.value) {
  login_status.value = "Nie ma takiego użytkownika";
  return;
}
if(check.password !== password .value) {
  login_status.value = "Błędne hasło";
  return;
}
login_status.value = "Logowanie..."
setTimeout(() => {
  document.getElementById("home_page").style.display='block';
  document.getElementById("backstage").style.display='block';
  document.getElementById("login_page").style.display='none';
},1000)
```
Thanks to that we remove the depth of the nesting and it's easier to follow :)

File: frontend/style.css [hunk 14]
```diff
@@ -269,11 +269,17 @@
 .drink_element {
   padding: 20px;
   width: 350px;
-  height: 600px;
+  height: 400px;
```
Comment (kfron) [inline frontend/style.css]: It is a bad idea to use `px` in styling, as it is not responsive. Imagine using this app on your phone, on a small screen or on an extra large screen. Read more about using `em`, `rem` and other values.
https://www.w3.org/Style/Examples/007/units.pl.html

File: frontend/drink_list_picker.js (hunk 1 above)
Comment (kfron) [reply frontend/drink_list_picker.js]: This is a problem in other files too, so remember to add an empty line at the end of every file (keep in mind that there should be ONLY ONE empty line at the end of the file)

**Review by kfron — CHANGES_REQUESTED** [review]

File: frontend/drink_list_picker.js (hunk 1 above)
Comment (kfron) [inline frontend/drink_list_picker.js]: I don't like that it auto-invokes itself, but I am not sure if that is a bad practice or not

Comment (Draganek) [reply frontend/drink_list_picker.js]: ok

File: frontend/drink_list_picker.js (hunk 2 above)
Comment (Draganek) [reply frontend/drink_list_picker.js]: done

File: frontend/drink_list_picker.js (hunk 3 above)
Comment (Draganek) [reply frontend/drink_list_picker.js]: done

File: frontend/drink_list_picker.js (hunk 4 above)
Comment (Draganek) [reply frontend/drink_list_picker.js]: fixed

File: frontend/index.html (hunk 5 above)
Comment (Draganek) [reply frontend/index.html:2]: ok

File: frontend/login_menu.js (hunk 10 above)
Comment (Draganek) [reply frontend/login_menu.js]: i will

File: frontend/login_menu.js (hunk 11 above)
Comment (Draganek) [reply frontend/login_menu.js]: ok

File: frontend/login_menu.js (hunk 12 above)
Comment (Draganek) [reply frontend/login_menu.js]: that was a good idea

File: frontend/login_menu.js (hunk 13 above)
Comment (Draganek) [reply frontend/login_menu.js]: nesting removed

File: frontend/style.css (hunk 14 above)
Comment (Draganek) [reply frontend/style.css]: ok, I will use em or rem or other better values

File: frontend/drink_list_picker.js (hunk 1 above)
Comment (Draganek) [reply frontend/drink_list_picker.js]: know it's done correctly

Comment (kfron) [reply frontend/drink_list_picker.js]: You left some files still without an empty line at the end.
style.css
index.html
Please look through all of the files in the project :)

File: frontend/drink_list_picker.js (hunk 4 above)
Comment (kfron) [reply frontend/drink_list_picker.js]: It's better, but I think we need a new convention for naming functions

File: frontend/login_menu.js [hunk 15]
```diff
@@ -1,40 +1,43 @@
...
-        login_status.value = "Nie ma takiego użytkownika"
-}
\ No newline at end of file
+        login_status.value = LoginStatuses.IN_PROGRESS;
+        setTimeout(() => {
+            document.getElementById("home_page").style.display = 'block';
+            document.getElementById("backstage").style.display = 'block';
+            document.getElementById("login_page").style.display = 'none';
+        }, 1000)
+    }
```
Comment (kfron) [inline frontend/login_menu.js:41]: There should be a `return` to stop the `for` loop

File: frontend/index.html [hunk 16]
```diff
@@ -1,143 +1,104 @@
...
+    </div>
 
-  <a style="color: white ; font-size: 40px;">
-    Tu będzie zajebista lista drinków do wybrania lol
-  </a>
+    <div id="drink_list_page" style="display:none">
+      <a style="color: white ; font-size: 40px;">
+        Tu będzie zajebista lista drinków do wybrania lol
+      </a>
+      <!-- <div id="show_drink" class="drink_window"></div> -->
```
Comment (kfron) [inline frontend/index.html]: Never leave commented code. Either remove it or uncomment if you want it

File: frontend/drink_list_picker.js [hunk 17]
```diff
@@ -1,11 +1,18 @@
...
-
-
-async function login_user() {
+async function all_drink_choices() {
+    let drink_list_page = document.getElementById("drink_list_page");
+    let created_list = document.getElementById("drink_list_show");
+    try {
+        drink_list_page.removeChild(created_list);
+    }
+    catch {
```
Comment (kfron) [inline frontend/drink_list_picker.js]: Why is there a try/catch? Do you expect an error? If so, why is there nothing in the catch statement?
And, if you do expect an error, you should try to be consistent and use a try/catch everywhere you remove a child :)

**Review by kfron — CHANGES_REQUESTED** [review]

**Review by kfron — APPROVED** [review]

**Pull request merged**